### PR TITLE
[WIP]: Run Travis-CI in Linux and OSX

### PR DESCRIPTION
**Pull request by rogeriopradoj**
WIP.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -22,3 +34,7 @@ before_script:
 
 after_failure:
     - sudo cat $(ls -r /tmp/php-build*.log | head -n 1)
+
+matrix:
+  allow_failures:
+    - os: osx
```
Comment (GrahamCampbell) [inline .travis.yml]: needs taking out of allow failures otherwise there's not much point

File: .travis.yml [hunk 2]
```diff
@@ -10,9 +14,17 @@ env:
 script: ./run-tests.sh $DEFINITION
 
 before_install:
-    - sudo apt-get update -qq
-    - sudo apt-get build-dep php5-cli
-    - sudo apt-get install -qq git-core libmcrypt-dev libjpeg-dev libreadline-dev
+    - echo $TRAVIS_OS_NAME
```
Comment (GrahamCampbell) [inline .travis.yml]: not needed

File: .travis.yml [hunk 3]
```diff
@@ -10,9 +14,17 @@ env:
...
+    # LINUX
```
Comment (GrahamCampbell) [inline .travis.yml]: i don't think these comments add any value, the if statements are pretty clear already

File: .travis.yml (hunk 1 above)
Comment (rogeriopradoj) [inline .travis.yml]: Hi, @GrahamCampbell , just a working in progress, ok? Before it is ready for get merged, I want to have it passing in osx either linux, however not changing allow_failures as you suggest.

As the project was just linux for so long, I guess it's not a problem to have it allowing failures for osx in this very begining .

IMHO it is like on PHP projects that putting HHVM in `allow_failures` for HHVM, cause the main point is that the code works on PHP versions, but, if works in HHVM again, is a bonus.

File: .travis.yml (hunk 2 above)
Comment (rogeriopradoj) [inline .travis.yml]: Really, not needed. Will change.

File: .travis.yml (hunk 3 above)
Comment (rogeriopradoj) [inline .travis.yml]: Really, not needed. Will change.

**Comment by hnw** [comment]
Hi. I noticed configure script in PHP 5.3.29 cannot find libssl on MacOSX 10.9+. It's because `DSA_get_default_method` is not in libssl, but in libcrypto on MacOSX 10.9+. So, we should specify `-lssl` explicitly, such as:

```
    - DEFINITION=5.3.29 LIBS="-lssl -lcrypto"
```

**Comment by rogeriopradoj** [comment]
Hi, @hnw , thanks for coming aboard! For 5.2.17 and 5.3.3 do you have suggestions? I getting hard time to fix build for these old versions...

**Comment by henriquemoody** [comment]
BTW, it needs a rebase because the current Travis configuration is using the new container based structure.

**Comment by rogeriopradoj** [comment]
Fixed in #342

**Pull request closed without merging**